codeSnippets/snippets/full-stack-task-manager/server/src/main/kotlin/com/example/ktor/full_stack_task_manager/Application.kt

```
@@ -0,0 +1,92 @@
+                    }
+                    call.respond(tasks)
+                } catch (ex: IllegalArgumentException) {
+                    call.respond(HttpStatusCode.BadRequest)
+                }
+            }
+            post {
+                try {
+                    val task = call.receive<Task>()
+                    repository.addOrUpdateTask(task)
+                    call.respond(HttpStatusCode.NoContent)
+                } catch (ex: IllegalStateException) {
```

detekt (1.23.7)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)